### PR TITLE
Ticket #937: Invitation does not work

### LINKED ISSUE #937: [Bug] Invitation to co-manage domain did not work
### Current Behavior

Added Gaby as co-manager of my domain (career-least-left.gov) in staging using the email she has linked to her Login.gov account. She was **not** instantly added as co-manager. Instead, she's pending in the "Invitations" queue. Even after she logged into the staging registrar, it never triggered to add her as a co-manager.

The functionality worked correctly for the other users I've tried (who also have Login.gov accounts). This is the only instance where it hasn't worked.

### Expected Behavior

Upon adding user with Login.gov account, they should immediately appear in the "Active users" table and have access to the domain.

### Steps to Reproduce

1. Add Gaby to an approved domain _(I'm not posting her email in this ticket for privacy reasons; reach out to her directly for email address)_
2. Note whether she is immediately added to the "Active users"table
3. Ask her to verify whether she has access to the domain in her account.

### Environment

_No response_

### Additional Context

_No response_

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #937

## Changes

- Fixes fixtures.py to look for emails in the right dictionary
- Changed how we're grabbing the email variable from the form to be more consistent with the rest of the doc

## Context for reviewers
Domain Invitation works by doing a .get on a user based off of their email address. In fixtures.py, all admin users do not have an associated email in ADMINS. This means that to the system, it looks like every admin does not exist when they do not have a defined email. This is easily fixable, but there was a bug with STAFF wherein while they do have emails, they are always null because we were looking in the ADMINS dictionary array and trying to grab their email, rather than the STAFF dict.

This bug is now fixed. This will work for @gabydisarli with her staff user. If she'd want her admin user to be invited to a domain, she would need to add her email to the ADMINS dict array. Gaby, would you want me to add you there? 

## Setup
Invite any user who has an email attribute in STAFF or in ADMINS. You can also add your own in fixtures.py if it is not there, which I did for test purposes.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Comment by zandercymatics** [comment]
Converting to a draft as there are other PRS pending currently

File: src/registrar/views/domain.py [hunk 1]
```diff
@@ -377,7 +377,7 @@ def _make_invitation(self, email_address):
 
     def form_valid(self, form):
         """Add the specified user on this domain."""
-        requested_email = form.cleaned_data["email"]
+        requested_email = form.cleaned_data.get("email", "")
```
Comment (abroddrick) [inline src/registrar/views/domain.py:380]: why empty string not None? and then log if None

**Comment by abroddrick** [comment]
@zandercymatics this solution wouldn't necessarily have solved the problem Katherine documented in that ticket unless Gaby logged in as analyst. Did you confirm that this was the case? ( I say as only her analyst account has her email).

**Comment by abroddrick** [comment]
Just dm-ed Gaby, for response on the above. 

@zandercymatics it looks like some merge conflict happened, is that correct?

**Comment by gabydisarli** [comment]
> Its worth it getting @gabydisarli's feedback here. Gaby, when testing admin functionality, do you typically use the same email that you used to log in with? Or do you use a separate email?


@abroddrick I have two accounts, 1 is a super user account that is gaby@truss.works and the other is an analyst account (for testing DjA that is gaby+1@truss.works

**Comment by gabydisarli** [comment]
> This bug is now fixed. This will work for @gabydisarli with her staff user. If she'd want her admin user to be invited to a domain, she would need to add her email to the ADMINS dict array. Gaby, would you want me to add you there?

@zandercymatics 
Yes, could you do that for me, please?

**Review by gabydisarli — APPROVED** [review]
- Confirmed I can now log in to DjA through Zander's sandbox using both my regular account and +1 account for analysts
- Received invitation emails to manage domains

File: src/registrar/views/domain.py (hunk 1 above)
Comment (zandercymatics) [reply src/registrar/views/domain.py:384]: Reverted this change, was not necessary

**Review by therealslimhsiehdy — APPROVED** [review]

**Pull request merged**